### PR TITLE
[FrameworkBundle] Allow multiple transitions with the same name

### PR DESCRIPTION
**Pull request by Padam87**
| Q             | A
| ------------- | ---
| Branch?       |  3.2
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #20794
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php [hunk 1]
```diff
@@ -409,6 +409,8 @@ private function registerWorkflowConfiguration(array $workflows, ContainerBuilde
 
             $transitions = array();
             foreach ($workflow['transitions'] as $transitionName => $transition) {
+                $transitionName = $transition['name'] ?: $transitionName;
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: I think we should normalise the values in the `Configuration` class already to ensure that an element with the `name` key is set.

**Comment by nicolas-grekas** [comment]
ping @lyrixx

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -295,11 +295,11 @@ private function addWorkflowSection(ArrayNodeDefinition $rootNode)
                                 ->end()
                             ->end()
                             ->arrayNode('transitions')
-                                ->useAttributeAsKey('name')
                                 ->isRequired()
                                 ->requiresAtLeastOneElement()
                                 ->prototype('array')
                                     ->children()
+                                        ->scalarNode('name')->defaultNull()->end()
```
Comment (lyrixx) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]:
```
->scalarNode('name')
    ->isRequired()
    ->cannotBeEmpty()
->end()
```

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -322,6 +322,14 @@ private function addWorkflowSection(ArrayNodeDefinition $rootNode)
                                         ->end()
                                     ->end()
                                 ->end()
+                                ->validate()
```
Comment (lyrixx) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]:
```
->beforeNormalization()
    ->always()
    ->then(function ($transitions) {
        // It's an indexed array, we let the validation occurs
        if (isset($transitions[0])) {
            return $transitions;
        }

        foreach ($transitions as $name => $transition) {
            $transition['name'] = $name;
            $transitions[$name] = $transition;
        }

        return $transitions;
    })
->end()
```

**Review by lyrixx — CHANGES_REQUESTED** [review]

**Comment by lyrixx** [comment]
Hello @Padam87 

Thanks for your PR. I added 2 comments about the code. Could you please update your code with my patch?

**Comment by stof** [comment]
We should still validate that you don't have 2 transitions with the same name and a common from node though.

**Comment by lyrixx** [comment]
> We should still validate that you don't have 2 transitions with the same name 

Yes, indedd

> and a common from node though.

Why? it's possible to have 2 transitions, with the same "from" name.

**Comment by stof** [comment]
@lyrixx your ``Yes indeed`` being in the middle of the ``and`` qualifying transitions makes me think that you misunderstood my comment entirely (your ``Yes indeed`` looks like you want to forbid exactly what this PR is about allowing to configure)

**Comment by lyrixx** [comment]
Oups, yes, you are right ;) Sorry.

**Comment by lyrixx** [comment]
But i'm not sure we should validate the "name + from" here. It should be done in the symfony/component validator instead.

**Comment by stof** [comment]
@lyrixx it indeed must be done in the workflow validator.
But it would not be bad to have an early validation in the DI class if it is easy to do (forbidding to configure an invalid workflow which would fail only later on). Validation during the compile time of the container would give early feedback without performance impact at runtime (as it happens at build time).

**Comment by lyrixx** [comment]
@stof All validations are already done in a compiler pass ;)

**Comment by stof** [comment]
then fine. I forgot that

**Comment by lyrixx** [comment]
Hello @Padam87

I'm sorry, I totally forgot this PR because it was not tagged with `Workflow`.

Thanks for this PR!

:+1:

**Comment by fabpot** [comment]
Thank you @Padam87.

**Pull request merged**